common/tier4_autoware_utils/src/vehicle/vehicle_state_checker.cpp

```diff
@@ -36,31 +36,31 @@ bool VehicleStopChecker::isVehicleStopped(const double stop_duration = 0.0) cons
     return false;
   }
 
+  constexpr double stop_velocity = 1e-3;
   const auto now = clock_->now();
-  const auto time_buffer = now - twist_buffer_.back().header.stamp;
-  if (time_buffer.seconds() < stop_duration) {
+
+  const auto time_buffer_front = now - twist_buffer_.back().header.stamp;
+  if (time_buffer_front.seconds() < stop_duration) {
     return false;
   }
 
+  const auto time_buffer_back = now - twist_buffer_.front().header.stamp;
```

Comment:
This naming (back = front) is very confusing, and this check is redundant. In the next loop, return false if the velocity of queue front is not zero. if not, break and return true.

Reply:
Oh... Sorry, correct implementation is fowllowing: :bow: 

```c++
bool VehicleStopChecker::isVehicleStopped(const double stop_duration = 0.0) const
{
  if (twist_buffer_.empty()) {
    return false;
  }

  constexpr double stop_velocity = 1e-3;
  const auto now = clock_->now();

  const auto time_buffer_back = now - twist_buffer_.back().header.stamp;
  if (time_buffer_back.seconds() < stop_duration) {
    return false;
  }

  const auto time_buffer_front = now - twist_buffer_.front().header.stamp;
  if (time_buffer_front.seconds() > stop_duration) {
    return twist_buffer_.front().twist.linear.x < stop_velocity;
  }

  // Get velocities within stop_duration
  for (const auto & velocity : twist_buffer_) {
    if (stop_velocity <= velocity.twist.linear.x) {
      return false;
    }

    const auto time_diff = now - velocity.header.stamp;
    if (time_diff.seconds() >= stop_duration) {
      break;
    }
  }

  return true;
}
```

Reply:
I fixed in https://github.com/autowarefoundation/autoware.universe/pull/896/commits/69e536418f25d18764234ecb27c87fb535a8127b.